hexrd/ui/interactive_template.py

```diff
@@ -149,11 +183,11 @@ def toggle_boundaries(self, show):
         self.redraw()
 
     def disconnect(self):
-        self.parent.mpl_disconnect(self.button_press_cid)
-        self.parent.mpl_disconnect(self.button_release_cid)
-        self.parent.mpl_disconnect(self.motion_cid)
-        self.parent.mpl_disconnect(self.key_press_cid)
-        self.parent.mpl_disconnect(self.button_drag_cid)
+        self.current_canvas.mpl_disconnect(self.button_press_cid)
+        self.current_canvas.mpl_disconnect(self.button_release_cid)
+        self.current_canvas.mpl_disconnect(self.motion_cid)
+        self.current_canvas.mpl_disconnect(self.key_press_cid)
+        self.current_canvas.mpl_disconnect(self.button_drag_cid)
```

Comment:
You might want to put `canvas = self.current_canvas` in the first line so that the `self.current_canvas` property only gets evaluated once. Same for `self.connect_translate_rotate()`.

Reply:
No longer applicable